client/src/sections/courses/LearningSection/index.tsx

```diff
@@ -35,23 +35,29 @@ const LearningSection: React.FC = () => {
     }
   }, [trainee, isLoading, dispatch]);
 
-  const handleSortOptionChange = (value: string): void => {
-    setSelectedSortOption(value);
+  useEffect(() => {
+    if (!isLoading && trainee) {
+      dispatch(sortTrainees(selectedSortOption));
+    }
+  }, [isLoading, dispatch]);
+
+  const handleSortOptionChange = (option: string): void => {
+    setSelectedSortOption(option);
+    setSelectedOption(option);
+    if (sortOption !== option) {
+      dispatch(resetTraineesList(option));
+    }
   };
 
+  useEffect(() => {
+    return () => {
+      dispatch(resetTraineesList(sortOption));
+    };
+  }, []);
+
```

Comment:
without this code block, it will have a bug see video below.
```js
  useEffect(() => {
    return () => {
      dispatch(resetTraineesList(sortOption));
    };
  }, []);
```


https://github.com/framgia/sph-lms/assets/81206481/e9a158b3-02d2-4b76-a00b-92b7875f29f8

Reply:
Okay, can we remove the other useEffect in line 38 and also sortTrainees in learnerSlice?

Reply:
Already removed